### DISCUSSION TITLE
blog/check-function

### DISCUSSION THREAD

**Discussion by giscus** [zenstackhq/zenstack-docs#360] [Comments]
# blog/check-function

This post introduces a typical pattern of access policy duplication in ZenStack schemas, and explains how the new `check` attribute function can help you keep your policies DRY.

https://zenstack.dev/blog/check-function

**Comment by jason-curtis** [comment]
I just applied `check()` to my models and I deleted so much code!! Thank you so much! It's a game-changer that puts ZenStack another step ahead of the other options in terms of DX.

And I can't wait to use `check()` on to-many relations as well.

As for syntax: `children?[child, check(child, 'read')])` is a bit more complex than would be ideal (why do you have to say `child` twice?). I think option `A` would be great (`checkSome`, `checkAll`, `checkNone`, etc.), or a simplified option `B`, something like `children?[check()]` or `check(children?)`.

Is there a GH issue that I can follow to participate in the discussion and find out when `check` is ready to use on to-many relations? (edit: I didn't see one so I filed https://github.com/zenstackhq/zenstack/issues/1688 ).

As for recursive `check`s, I don't need those for the time being but I appreciate the thought you've put into it.
